src/NMFMerge.jl

```diff
@@ -148,8 +148,12 @@ function solve_remix(S, T, id1, id2)
     b = sqrt(τ^2/4-δ)
     λ_max = τ/2+b
     λ_min = δ/λ_max
-    ξ = (h1h1-h2h2+2b)/((h1h2+c*h2h2)*2)
-    u = (ξ, 1)./sqrt(1+2ξ*c+ξ^2)
+    if c == 0 && h1h2 == 0
+        u = h1h1 >= h2h2 ? (one(c), zero(c)) : (zero(c), one(c))
+    else
+        ξ = (h1h1-h2h2+2b)/((h1h2+c*h2h2)*2)

```

Comment:
```suggestion
    den = (h1h2+c*h2h2)*2
    if iszero(den)
        u = h1h1 >= h2h2 ? (one(c), zero(c)) : (zero(c), one(c))
    else
        ξ = (h1h1-h2h2+2b)/den
```
This is slightly more robust, as there are extreme cases where `x*y == 0` even if `x` and `y` are not:
```julia
julia> x = 1e-200
1.0e-200

julia> x^2
0.0
```